libraries/chain/controller.cpp

```diff
@@ -804,7 +804,9 @@ struct controller_impl {
                EOS_ASSERT( false, block_validate_exception, "encountered unexpected receipt type" );
             }
 
-            if( trace && trace->except ) {
+            bool transaction_failed =  trace && trace->except;
+            bool transaction_can_fail = receipt.status == transaction_receipt_header::hard_fail && receipt.trx.contains<transaction_id_type>();
+            if( transaction_failed && !transaction_can_fail) {
```

Comment:
Let's add a method to transaction_receipt for determination of delayed/deferred transactions and unit tests for next release.

Reply:
If the number of delayed transactions increases, this case is meaningless. My opinion is that the handling of delayed transactions has to change.

Reply:
Just  PR Three lines of code can fix bug?

Reply:
This reads funny (very unintuitive):
transaction_failed && !transaction_can_fail